### PR TITLE
Allow reverify inventories

### LINKED ISSUE #2322: public UpdateExtensibleWitnessWhiteList and verify method in ExtensiblePayload
https://github.com/neo-project/neo/blob/6cb8cc951ee22415a531fca7811f76c3966771f7/src/neo/Ledger/Blockchain.cs#L384-L406

https://github.com/neo-project/neo/blob/6cb8cc951ee22415a531fca7811f76c3966771f7/src/neo/Network/P2P/Payloads/ExtensiblePayload.cs#L93-L101

as required for https://github.com/neo-project/neo-modules/pull/517 used by DBFT plugin.

### PR DESCRIPTION
**Pull request by erikzhang**
Close #2322 close #2325

### CODE REVIEW

File: src/neo/Ledger/TransactionRouter.cs [hunk 1]
```diff
@@ -16,12 +19,8 @@ public TransactionRouter(NeoSystem system)
...
         protected override void OnReceive(object message)
         {
-            if (message is not Transaction tx) return;
-            system.Blockchain.Tell(new Blockchain.PreverifyCompleted
-            {
-                Transaction = tx,
-                Result = tx.VerifyStateIndependent(system.Settings)
-            }, Sender);
+            if (message is not Preverify preverify) return;
+            system.Blockchain.Tell(new PreverifyCompleted(preverify.Transaction, preverify.Relay, preverify.Transaction.VerifyStateIndependent(system.Settings)), Sender);
```
Comment (shargon) [inline src/neo/Ledger/TransactionRouter.cs:23]: How we will know that the message it's our reverification, in order to not do it again?

Comment (erikzhang) [reply src/neo/Ledger/TransactionRouter.cs:23]: What do you mean? This is `Preverify`, not `Reverify`...

Comment (shargon) [reply src/neo/Ledger/TransactionRouter.cs:23]: I mean, the answer is the same for verify and reverify, so we cannot distinguish asynchronously.

Comment (erikzhang) [reply src/neo/Ledger/TransactionRouter.cs:23]: Why do you need to distinguish it? We need it in dBFT plugin to verify the payloads in `RecoveryMessage`.

**Review by shargon — APPROVED** [review]

**Pull request merged**